src/rust/runtime/memory/mod.rs

```diff
@@ -52,6 +52,28 @@ pub trait MemoryRuntime {
         })
     }
 
+    /// Converts a scatter-gather array into a buffer.
+    fn into_buf(&self, sga: &demi_sgarray_t) -> Result<DemiBuffer, Fail> {
+        // Check arguments.
+        // TODO: Drop this check once we support scatter-gather arrays with multiple segments.
+        if sga.sga_numsegs != 1 {
+            return Err(Fail::new(libc::EINVAL, "demi_sgarray_t has invalid segment count"));
```

Comment:
We could enhance this message to state that multiple segments are not currently supported.